tests/repository_simulator.py

```diff
@@ -354,21 +355,60 @@ def add_delegation(
         else:
             delegator = self.md_delegates[delegator_name].signed
 
+        if (
+            delegator.delegations is not None
+            and delegator.delegations.succinct_roles is not None
+        ):
+            raise ValueError("Can't add a role when succinct_roles is used")
+
         # Create delegation
         if delegator.delegations is None:
-            delegator.delegations = Delegations({}, {})
+            delegator.delegations = Delegations({}, roles={})
+
+        assert delegator.delegations.roles is not None
```

Comment:
I do this because of mypy. Otherwise, it gives a warning:
`error: Unsupported target for indexed assignment ("Optional[Dict[str, DelegatedRole]]")  [index]`